selftests/check.py

```diff
@@ -575,8 +575,18 @@ def create_suites(args):  # pylint: disable=W0621
         suites.append(TestSuite.from_config(config_check, "jobs"))
 
     if args.dict_tests['functional']:
-        config_check['resolver.references'] = ['selftests/functional/']
-        suites.append(TestSuite.from_config(config_check, "functional"))
+        functional_path = os.path.join('selftests', 'functional')
+        references = glob.glob(os.path.join(functional_path, 'test*.py'))
+        references.extend([os.path.join(functional_path, 'utils'),
+                           os.path.join(functional_path, 'plugin')])
+        config_check['resolver.references'] = references
+        suites.append(TestSuite.from_config(config_check,
+                                            "functional-parallel"))
+
+        config_check['resolver.references'] = ['selftests/functional/serial/']
+        config_check['nrunner.max_parallel_tasks'] = 1
+        suites.append(TestSuite.from_config(config_check, "functional-serial"))
```

Comment:
IIUC, here you are using the same dict for two different suites. Due to the dict nature, one suite could update the dict and modify on the second suite. So, just to be a little bit safer, maybe a `copy` or `deep copy` here?

Reply:
Yes, I can do that. I was reusing the approach from other types of tests, but I can update them as well.